stdlib/fs.gr

```diff
@@ -0,0 +1,1048 @@
+ * @returns `Ok(void)` if the operation succeeds, `Err(err)` if a file system error is encountered
+ * 
+ * @example Fs.remove(Path.fromString("file.txt")) // removes a file
+ * @example Fs.remove(Path.fromString("dir"), removeMode=Fs.RemoveEmptyDirectory) // removes an empty directory
+ * @example Fs.remove(Path.fromString("dir"), removeMode=Fs.RemoveRecursive) // removes the directory and its contents
+ * 
+ * @since v0.6.5
+ */
+provide let remove = (baseDirPath=None, path, removeMode=RemoveFile) => {
+  let (baseDirPath, path) = getBaseDirAndPath(baseDirPath, path)
+  match (removeMode) {
+    RemoveFile =>
+      fileOp(
+        baseDirPath,
+        Unlink,
+        baseDirFd => File.pathUnlink(baseDirFd, Path.toString(path))
+      ),
+    RemoveEmptyDirectory =>
+      fileOp(
+        baseDirPath,
+        RmDir,
+        baseDirFd => File.pathRemoveDirectory(baseDirFd, Path.toString(path))
+      ),
+    RemoveRecursive =>
+      wrappingFileOp(
+        baseDirPath,
+        RmrfDir,
+        baseDirFd => removeRecursive(baseDirFd, Path.toString(path))
+      ),
+  }
+}
+
+/**
+ * Reads the contents of a directory.
+ * 
+ * @param baseDirPath: The path to the directory in which resolution should begin
+ * @param path: The path to the directory to read
+ * @returns `Ok(contents)` containing the directory contents or `Err(err)` if a file system error is encountered
+ * 
+ * @since v0.6.5
+ */
+provide let readDir = (baseDirPath=None, path) => {
+  let path = joinPathOnBaseDir(baseDirPath, path)
+  let result = fileOp(path, ReadDir, File.fdReaddir)
+  Result.map(entries => {
+    let entries = Array.map(
+      (entry: File.DirectoryEntry) =>
+        { name: entry.path, fileType: wrapFileType(entry.filetype) },
+      entries
+    )
+    Array.toList(entries)
+  }, result)
+}
+
+/**
+ * Creates a new empty directory at the given path.
+ * 
+ * @param baseDirPath: The path to the directory in which resolution should begin
+ * @param path: The path to create the new directory, relative to the base directory
+ * @returns `Ok(void)` if the operation succeeds, `Err(err)` if a file system error is encountered
+ * 
+ * @since v0.6.5
+ */
+provide let makeDir = (baseDirPath=None, path) => {
+  let (baseDirPath, path) = getBaseDirAndPath(baseDirPath, path)
+  fileOp(
+    baseDirPath,
+    MakeDir,
+    dirFd => File.pathCreateDirectory(dirFd, Path.toString(path))
+  )
+}
+
+/**
+ * Creates a new symbolic link with the given contents.
+ * 
+ * @param linkContents: The path to store into the link
+ * @param targetBaseDirPath: The path to the directory in which the target path resolution starts
+ * @param targetPath: The path to the target of the link
+ * @returns `Ok(void)` if the operation succeeds, `Err(err)` if a file system error or relativization error is encountered
+ * 
+ * @since v0.6.5
+ */
+provide let makeSymlink = (linkContents, targetBaseDirPath=None, targetPath) => {
```

Comment:
I think this should be called `createSymlink`.

Reply:
Ok going to rename `makeDir` to `createDir` as well for consistency